test-project.patch

```diff
@@ -0,0 +1,84 @@
++            compile.getTargetCompatibility().set("11");
+             compile.getOptions().getCompilerArgs().add("-proc:none");
+             compile.doLast(new Action<Task>() {
+                 @Override
+@@ -27,7 +27,7 @@
+             });
+         });
+         project.getTasks().withType(ProcessResources.class).configureEach(processResources -> {
+-            processResources.setDestinationDir(project.getLayout().getBuildDirectory().dir("new-resources").get().getAsFile());
++            processResources.getDestinationDir().set(project.getLayout().getBuildDirectory().dir("new-resources"));
+         });
+
+         project.getDependencies().constraints(constraints -> {
+Index: test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
+diff --git a/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle b/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle
+--- a/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle	(revision c88205e9ca573c625403202c42a9e8f64656e431)
++++ b/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle	(date 1652965995179)
+@@ -8,7 +8,7 @@
+ tasks.withType(AbstractCompile).configureEach {
+     targetCompatibility = "11"
+     sourceCompatibility = "11"
+-    classpath += files()
++    classpath.from(files())
+     doLast {
+         println("Hello from Groovy plugin, classpath length: ${classpath.files.size()}")
+     }
+Index: test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
+diff --git a/test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt b/test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt
+--- a/test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt	(revision c88205e9ca573c625403202c42a9e8f64656e431)
++++ b/test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt	(date 1652965995189)
+@@ -9,7 +9,7 @@
+ class TestKotlinPlugin : Plugin<Project> {
+     override fun apply(project: Project) {
+         project.tasks.withType<ProcessResources>().configureEach {
+-            destinationDir = project.layout.buildDirectory.dir("new-resources").get().asFile
++            destinationDir = project.layout.buildDirectory.dir("new-resources")
```

Comment:
This change isn't really necessary for the upgrade, though it improves things, right?

Reply:
I added comments to explain each change. I wonder if we should add comments in places where we did _not_ have to change things..?

Reply:
I think the comments in the places where we didn't need to change things would be good as well.